File: src/service/divide_server.cpp
#include "ros/ros.h"
#include "ros_basics_tutorials/DivideTwoInts.h"

bool divide(ros_basics_tutorials::DivideTwoInts::Request &req, ros_basics_tutorials::DivideTwoInts::Response &res)
{
    res.result = (double)req.a / req.b;
    ROS_INFO("request: x=%ld, y=%ld", (long int)req.a, (long int)req.b);
    ROS_INFO("sending response: [%lf]", (double)res.result);
    return true;
}
int main(int argc, char **argv)
{
    ros::init(argc, argv, "divide_two_ints_server");
    ros::NodeHandle n;

    ros::ServiceServer service = n.advertiseService("divide_two_ints", divide);
    ROS_INFO("Ready to divide");
    ros::spin();

    return 0;
}

File: src/service/divide_server.py
#!/usr/bin/env python

from ros_basics_tutorials.srv import DivideTwoInts
from ros_basics_tutorials.srv import DivideTwoIntsRequest
from ros_basics_tutorials.srv import DivideTwoIntsResponse
import rospy

import time


def handle_divide_two_inits(req):
    print("Returning [%s / %s = %s]" % (req.a, req.b, (req.a/req.b)))
    time.sleep(5)
    response = DivideTwoIntsResponse(req.a/req.b)
    return response


def divide_two_ints_server():
    rospy.init_node('divide_two_ints_server')
    s = rospy.Service('divide_two_ints', DivideTwoInts,
                      handle_divide_two_inits)
    print("Ready to divide tow ints.")
    rospy.spin()


if __name__ == "__main__":
    divide_two_ints_server()
